### PR TITLE
Provide location parameter to KS test, address issue #39

### PR DESCRIPTION
**Pull request by eastjames**
This pull request fixes the KS Test by providing the location (fixed) parameters and resolves issue #39.

### CODE REVIEW

**Comment by georgebv** [comment]
Thank you. A few requests:

- let's keep naming consistent and call the variable `fixed_parameters`
- please update docstring for the new parameter
- update tests

**Comment by eastjames** [comment]
Added a new commit to address these points

**Comment by georgebv** [comment]
Thank you James. I have reviewed the PR and I think I realized that I previously misunderstood the problem. A fix still needs to be pushed however, let me elaborate.

I reviewed ks test implementation and I think it's correct. The idea is that you pass sample (extremes), distribution, and fit parameters. Fit parameters are used to convert distribution's CDF function of quantiles + fit parameters to a function of only quantiles - this produces a function of only quantiles. Fit parameters has all parameters (free and fixed). Because of this, we shouldn't be modifying the `KolmogorovSmirnov` class itself.

Regarding the original issue. Pyextremes `EVA` class works by doing the following:

1. Extract extremes
2. Transform extremes to standard form expected by distribution (e.g. minima to pseudo-maxima) - this is done with the `ExtremesTransformer` class
3. Fit distribution to transformed extremes - this is where terminology goes a bit haywire (my bad). Distribution has `fit_parameters` attribute, which has estimated parameters, and `fixed_parameters`, which has frozen (e.g., `floc`) parameters.
4. Get extreme values by calculating ISF for given exceedance probability and then transforming the calculated value back (remember step 2) to original extreme value domain

Where the issue truly lies is how the `EVA.test_ks` method is implemented. To summarize:

- `KolmogorovSmirnov` expects a sample, distribution, and all (free and fixed) parameters
- `EVA` has raw extremes and transformed extremes (we must use transformed because that's what fit parameters are for)
- `EVA.model` has fit parameters which are only free parameters (bad choice of terminology on my end)

What we need to do is to change the `EVA.test_ks` method to pass the following to to `KolmogorovSmirnov`:

- `self.extremes_transformer.transformed_extremes` to `extremes`
- `self.model.fit_parameters` (this is the same as `self.model.distribution.mle_parameters`) combined with `self.model.distribution.fixed_parameters` to `fit_parameters` - this way we will be passing all (free and fixed) parameters

My apologies for sending you on a wild goose chase earlier. Please let me know if you would be willing to revert the changes and update code.

**Comment by eastjames** [comment]
George, thank you for the clarification. I see why changes to `KolmogorovSmirnov` class aren't needed -- I've changed the PR to only have the necessary changes to `EVA.test_ks`. 

Using the term `fit_parameters` to mean (very slightly) different things in different places of the code is a bit confusing. However, you've already noted this and I think it is outside the scope of this PR!

File: src/pyextremes/eva.py [hunk 1]
```diff
@@ -217,9 +217,15 @@ def AIC(self) -> float:
...
-            extremes=self.extremes,
+            extremes=self.extremes_transformer.transformed_extremes,
             distribution=self.distribution.distribution,
-            fit_parameters=self.model.fit_parameters,
+            fit_parameters={
+                **self.model.fit_parameters,
+                **{
+                    'loc' if k=='floc' else
+                    k: v for k, v in model.distribution.fixed_parameters.items()
+                }
```
Comment (georgebv) [inline src/pyextremes/eva.py:227]: I think you meant `self.model`.

As for stripping `f` from fixed parameters like `floc` you can do something like this (we know that all fixed parameters in `scipy` must have prefix `f`):

https://github.com/georgebv/pyextremes/blob/a0b4baab5f1263691e6e32f3e26c14877d9a2f39/src/pyextremes/models/distribution.py#L99-L101

or you can just use the property directly like so:

https://github.com/georgebv/pyextremes/blob/a0b4baab5f1263691e6e32f3e26c14877d9a2f39/src/pyextremes/models/distribution.py#L218-L223

**Review by georgebv — CHANGES_REQUESTED** [review]
Thank you. Once you fix the commented section we should be good to merge.

Also I am not sure if tests will necessarily pass - data used in unit tests is actual data from a tidal gauge in NYC so it may fail to pass the KS test even though one would expect it would. If it doesn't, we can use synthetic data (sample from distribution) to make sure good samples pass and bad samples (sampled from different distribution) fail.

File: src/pyextremes/eva.py (hunk 1 above)
Comment (eastjames) [reply src/pyextremes/eva.py]: Thank you -- the PR is updated to use `self._fixed_parameters` directly

**Pull request merged**